product_analytic/models/account_move.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2015 Akretion (http://www.akretion.com/) - Alexis de Lattre
+# Copyright 2016 Antiun Ingeniería S.L. - Javier Iniesta
+# Copyright 2017 Tecnativa - Luis Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+INV_TYPE_MAP = {
+    "out_invoice": "income",
+    "out_refund": "income",
+    "in_invoice": "expense",
+    "in_refund": "expense",
+}
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        res = super()._onchange_product_id()
+        inv_type = self.move_id.move_type
+        if self.product_id and inv_type:
+            ana_accounts = (
+                self.product_id.product_tmpl_id._get_product_analytic_accounts()
+            )
+            ana_account = ana_accounts[INV_TYPE_MAP[inv_type]]
+            self.analytic_account_id = ana_account.id
+        return res
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            inv_type = self.env.context.get("inv_type", "out_invoice")
```

Comment:
In v14 this context is always false, leading to setting by default the product's income analytic account in all the new account.move.line. It seems better to use directly the account.move's `move_type`
```suggestion
            inv_type = self.env["account.move"].browse([vals.get("move_id")]).move_type
```